projects/smoldot-browser-demo/src/view.ts

```diff
@@ -0,0 +1,133 @@
+  #syncMessage: HTMLElement | undefined;
+
+  constructor(options: Options, model: Model) {
+    this.#options = options;
+    this.#model = model;
+    const container = document.getElementById(this.#options.containerId);
+    if (container === null) {
+      throw Error('Could not find the container. Did you change the Html?');
+    }
+    this.#container = container;
+  }
+
+  private timeElapsed(from: number, till: number) {
+    return ((till - from)/1000).toFixed(2);
+  }
+
+  private timestampHtml(withTime?:boolean): HTMLElement {
+    const timestampDiv =  document.createElement('time');
+    if (!withTime) {
+      return timestampDiv;
+    }
+
+    const time = performance.now();
+    timestampDiv.appendChild(document.createTextNode(
+      `${new Date().toLocaleTimeString()} (${this.timeElapsed(this.#model.loadTime, time)}s)`
+    ));
+
+    return timestampDiv
+  }
+
+  private messageHtml(message: string, withTime?: boolean): HTMLElement {
+    const messageDiv = document.createElement('div');
+    messageDiv.classList.add('message');
+    messageDiv.appendChild(this.timestampHtml(withTime))
+    messageDiv.appendChild(document.createTextNode(message));
+    return messageDiv;
+  }
+
+  private errorHtml(message: string): HTMLElement {
+    const messageDiv = document.createElement('div');
+    messageDiv.classList.add('message');
+    messageDiv.classList.add('error');
+    messageDiv.appendChild(document.createTextNode(message));
+    return messageDiv;
+  }
+
+  private displayMessage(message: any) {
+    this.#container.appendChild(message);
+  }
+
+  error(error: Error) {
+    this.displayMessage(this.errorHtml(error.message));
+    throw error;
+  };
+
+  log(message: string, withTime?: boolean) {
+    this.displayMessage(this.messageHtml(message, withTime));
+  }
+
+  #insertAtTopOfContainer = (el: HTMLElement) => {
+    if (this.#container.firstChild == null) {
+      this.#container.appendChild(el);
+    } else {
+      this.#container.insertBefore(el, this.#container.firstChild)
+    }
+  }
+
+  #ensureClassOn = (el: HTMLElement, className: string) => {
+    if (!el.classList.contains(className)) {
+      el.classList.add(className);
+    }
+  }
+
+  showSyncing() {
+    if (!this.#syncMessage) {
+      // mesage container
+      const syncState = document.createElement('div');
+      syncState.classList.add('message');
+
+      //contents - empty timestamp and pulsing message
+      syncState.appendChild(this.timestampHtml())
+      const syncMessage = document.createElement('em');
+      syncMessage.classList.add('pulse');
+      syncMessage.innerHTML = `${emojis.chain} Chain is syncing...`;
+      syncState.appendChild(syncMessage);
+
+      this.#syncMessage = syncMessage;
+      this.#syncState = syncState;
+      this.#insertAtTopOfContainer(this.#syncState);
+    } else {
+      // REM: Cover case that we change from synced state back to syncing.
+      // Would this ever happen?
```

Comment:
I think it could easily happen for example in cases like:
- Internet connection is interrupted
- Computer is send to sleep, sync is interrupted, you wake up the computer

Reply:
Ah yeah of course!  Let me account for these scenarios